docs/standard/garbage-collection/implementing-dispose.md

```diff
@@ -1,158 +1,158 @@
 ---
 title: "Implementing a Dispose method"
-ms.date: "04/07/2017"
+ms.date: 05/05/2020
 ms.technology: dotnet-standard
-dev_langs: 
+dev_langs:
   - "csharp"
   - "vb"
-helpviewer_keywords: 
+helpviewer_keywords:
   - "Dispose method"
   - "garbage collection, Dispose method"
 ms.assetid: eb4e1af0-3b48-4fbc-ad4e-fc2f64138bf9
 ---
 
 # Implementing a Dispose method
 
-You implement a <xref:System.IDisposable.Dispose%2A> method to release unmanaged resources used by your application. The .NET garbage collector does not allocate or release unmanaged memory.  
-  
-The pattern for disposing an object, referred to as a [dispose pattern](implementing-dispose.md), imposes order on the lifetime of an object. The dispose pattern is used only for objects that access unmanaged resources, such as file and pipe handles, registry handles, wait handles, or pointers to blocks of unmanaged memory. This is because the garbage collector is very efficient at reclaiming unused managed objects, but it is unable to reclaim unmanaged objects.  
-  
-The dispose pattern has two variations:  
-  
-- You wrap each unmanaged resource that a type uses in a safe handle (that is, in a class derived from <xref:System.Runtime.InteropServices.SafeHandle?displayProperty=nameWithType>). In this case, you implement the <xref:System.IDisposable> interface and an additional `Dispose(Boolean)` method. This is the recommended variation and doesn't require overriding the <xref:System.Object.Finalize%2A?displayProperty=nameWithType> method.  
-  
+You implement a <xref:System.IDisposable.Dispose%2A> method to release unmanaged resources used by your application. The .NET garbage collector does not allocate or release unmanaged memory.
+
+The pattern for disposing an object, referred to as the [dispose pattern](implementing-dispose.md), imposes order on the lifetime of an object. The dispose pattern is used only for objects that access unmanaged resources, such as file and pipe handles, registry handles, wait handles, or pointers to blocks of unmanaged memory. This is because the garbage collector is very efficient at reclaiming unused managed objects, but it is unable to reclaim unmanaged objects.
+
+The dispose pattern has two variations:
+
+- You wrap each unmanaged resource that a type uses in a safe handle (that is, in a class derived from <xref:System.Runtime.InteropServices.SafeHandle?displayProperty=nameWithType>). In this case, you implement the <xref:System.IDisposable> interface and an additional `Dispose(Boolean)` method. This is the recommended variation and doesn't require overriding the <xref:System.Object.Finalize%2A?displayProperty=nameWithType> method.
```

Comment:
📝 For years I was a big supporter of this pattern. However, over time I have come to strongly believe that this guidance is needlessly complex for application developers (I believe @davkean is the one who originally challenged by thinking here).

When safe handles are used, the guidance can be modified as follows:

1. Implement `IDisposable.Dispose` as a virtual method.
2. _Do not_ add a finalizer to this type, or to any type derived from it.

This situation is readily detected at compilation time (even in metadata), and can therefore be enforced at build time by analyzers. I refer to this pattern as the "disposable pattern", and to the previous one based on `Dispose(bool)` as the "legacy disposable pattern".

Reply:
Hi @sharwell,
Thanks for your feedback here, much appreciated. This is currently a draft and not an official pull request yet. I will take this all into consideration - and look forward to collaborating on this. Thanks again!